lib/HLSL/DxilScalarizeVectorIntrinsics.cpp

```diff
@@ -60,28 +61,38 @@ class DxilScalarizeVectorIntrinsics : public ModulePass {
       if (!HlslOP->GetOpCodeClass(Func, OpClass))
         continue;
 
-      bool NeedsRewrite = (Func->getReturnType()->isVectorTy() ||
+      bool CouldRewrite = (Func->getReturnType()->isVectorTy() ||
```

Comment:
Is this check actually needed? I think we're effectively checking the same things again in the for loop. 
I think it would be functionally equivalent and similar performance to just remove the check for CouldRewrite?